File: KCD/HMCombinedBattleCommand.m
//
//  HMCombinedBattleCommand.m
//  KCD
//
//  Created by Hori,Masaki on 2014/08/16.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

#import "HMCombinedBattleCommand.h"

#import "HMCalculateDamageCommand.h"
#import "HMDropShipHistoryCommand.h"
#import "HMGuardShelterCommand.h"


@implementation HMCombinedBattleCommand
+ (void)load
{
	static dispatch_once_t onceToken;
	dispatch_once(&onceToken, ^{
		[HMJSONCommand registerClass:self];
	});
}

+ (BOOL)canExcuteAPI:(NSString *)api
{
	if([api isEqualToString:@"/kcsapi/api_req_combined_battle/battle"]) return YES;
	if([api isEqualToString:@"/kcsapi/api_req_combined_battle/airbattle"]) return YES;
	if([api isEqualToString:@"/kcsapi/api_req_combined_battle/battle_water"]) return YES;
	if([api isEqualToString:@"/kcsapi/api_req_combined_battle/midnight_battle"]) return YES;
	if([api isEqualToString:@"/kcsapi/api_req_combined_battle/sp_midnight"]) return YES;
	if([api isEqualToString:@"/kcsapi/api_req_combined_battle/battleresult"]) return YES;
	
	return NO;
}

- (id)init
{
	self = [super initWithCommands:
			[HMDropShipHistoryCommand new],
			[HMCalculateDamageCommand new],
			[HMGuardShelterCommand new],
			nil];
	return self;
}
@end

File: KCD/HMCombinedBattleCommand.swift
//
//  HMCombinedBattleCommand.swift
//  KCD
//
//  Created by Hori,Masaki on 2015/01/13.
//  Copyright (c) 2015年 Hori,Masaki. All rights reserved.
//

import Cocoa

class HMCombinedBattleCommand: HMCompositCommand, NSCopying {
	required override init(commandArray: [AnyObject]) {
		super.init(commandArray: commandArray)
	}
	
	override class func load() {
		// Mavericksにおいてload()のタイミングでNSArrayを生成することが出来ないため、遅延させる
		dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(1 * Double(NSEC_PER_MSEC))), dispatch_get_main_queue()) {
			HMJSONCommand.registerInstance(self(commandArray: self.commands()))
		}
	}
	override class func canExcuteAPI(api: String) -> Bool {
		if api == "/kcsapi/api_req_combined_battle/battle" { return true }
		if api == "/kcsapi/api_req_combined_battle/airbattle" { return true }
		if api == "/kcsapi/api_req_combined_battle/battle_water" { return true }
		if api == "/kcsapi/api_req_combined_battle/midnight_battle" { return true }
		if api == "/kcsapi/api_req_combined_battle/sp_midnight" { return true }
		if api == "/kcsapi/api_req_combined_battle/battleresult" { return true }
		return false
	}
	
	override func copyWithZone(zone: NSZone) -> AnyObject {
		return self.dynamicType.init(commandArray: self.dynamicType.commands())
	}
	
	class func commands() -> [AnyObject] {
		return [HMCalculateDamageCommand()]
	}
}
